snowplow-tracker/src/androidTest/java/com/snowplowanalytics/snowplow/tracker/SubjectTest.java

```diff
@@ -114,5 +130,27 @@ public void testSubjectUserIdCanBeUpdated() {
         assertNull(tracker.getSubject().getUserId());
     }
 
+    @Test
+    public void testAnonymisesUserIdentifiers() {
+        Subject subject = getSubject();
+        subject.setUserId("uid-test");
+        subject.setDomainUserId("duid-test");
+        subject.setNetworkUserId("nuid-test");
+        subject.setIpAddress("127.0.0.1");
+        assertNull(subject.getSubject(true).get("uid"));
```

Comment:
Question: why did you call `subject.getSubject()` for each assertion, instead of saving the `subject.getSubject()` map as a variable and then using that? Is it to avoid making a variable?

Reply:
It's just my personal preference for writing tests – I try to avoid introducing variables in tests because it makes them slightly more difficult to read as one has to look for how the variables are intialized instead of reading the test from top to bottom. It doesn't make much difference in this case though, here it's mostly out of habit.